src/Microsoft.Health.Fhir.Core/Features/Persistence/IFhirDataStore.cs

```diff
@@ -20,5 +20,7 @@ public interface IFhirDataStore
         Task<ResourceWrapper> GetAsync(ResourceKey key, CancellationToken cancellationToken);
 
         Task HardDeleteAsync(ResourceKey key, CancellationToken cancellationToken);
+
+        Task<ResourceWrapper> ReplaceAsync(ResourceWrapper resourceWrapper, WeakETag weakETag, CancellationToken cancellationToken);
```

Comment:
For the purposes of Reindex, we don't need to replace the entire resource.  Just update the search index values, and the searchParamHash.  For CosmosDB, since it is all one document, I imagine we simply update the entire thing.  But for other persistence providers we could only send the value that we want to the DB.  Should we name this UpdateMetadataAsync?  Or UpdateSearchParams?

Reply:
I think even if we do change the method name, the input parameter will still be a `ResourceWrapper` if we want to maintain consistency across the different persistence providers right? I was thinking of `Replace` as more of a generic api call that is part of the datastore. And that it could potentially be used for other use cases also. I don't have a strong opinion on this though. I noticed that there are other methods now in `IFhirDataStore` that are specific to `SearchParameter`. I will update this method name also to reflect that.

Reply:
Update name to UpdateSearchParameterForResourceAsync

Reply:
The searchIndex is what is stored on the wrapper. Good to read the context above, I'd prefer something along the lines of `UpdateSearchIndexForResourceAsync`. We also introduced a SearchParameterStatusDataStore so I'd prefer to diverge from this naming a little.